### PR TITLE
[Doc] add limits in array and align en

### PR DESCRIPTION
**Pull request by evelynzhaojie**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3

### CODE REVIEW

**Review by hellolilyliuyi — DISMISSED** [review]

File: docs/sql-reference/sql-statements/data-types/Array.md [hunk 1]
```diff
@@ -1,108 +1,207 @@
 # ARRAY
 
-## Background
+ARRAY, as an extended type of databases, is supported in various database systems such as PostgreSQL, ClickHouse, and Snowflake. ARRAY is widely used in scenarios such as A/B tests, user tag analysis, and user profiling. StarRocks supports multidimensional array nesting, array slicing, comparison, and filtering.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-types/Array.md:3]: image

**Review by EsoragotoSpirit — DISMISSED** [review]

**Review by amber-create — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**